ebmdatalab/maps.py

```diff
@@ -1,131 +1,142 @@
-
-    # Add names and if it's London. Note the names in ccg_for_map must
-    # match names in the CCG geojson, as that doesn't include codes at
-    # the momemt
-    names = pd.read_csv(data_dir / "ccg_for_map.csv")
-
-    # Check we know about all the codes in the input data
-    diff = np.setdiff1d(df["pct"], names["code"])
-    if len(diff) > 0:
-        raise BaseException(
-            "Data contains CCG codes we don't know about: {}".format(diff)
-        )
-
-    df = df.merge(names[["code", "name", "is_london"]], left_on="pct", right_on="code")
-    df = df.set_index("name")
-
-    # Load map data
-    cartogram_suffix = ""
-    if cartogram:
-        cartogram_suffix = "_cartogram"
-    if map_year:
-        map_file = data_dir / "ccgs{}_{}.json".format(cartogram_suffix, map_year)
-    else:
-        map_file = sorted(
-            glob.glob(str(data_dir / "ccgs{}_2*.json".format(cartogram_suffix)))
-        )[-1]
-    ccgs = gpd.read_file(map_file)
-    # Normalise names to match `ccg_fo_map` format (above)
-    ccgs["name"] = ccgs["name"].str.upper()
-    ccgs = ccgs.set_index("name")
-    # Remove ones without geometry - these are (usually) federations
-    # rather than individual CCGs
-    ccgs = ccgs[~ccgs["geometry"].isnull()]
-
-    # Check we can map all the CCGs named in the input data
-    diff = np.setdiff1d(df.index, ccgs.index)
-    if len(diff) > 0:
-        raise BaseException("Data contains CCG names we can't map: {}".format(diff))
-
-    # Join map with data
-    gdf = ccgs.join(df, rsuffix="_orig")
-
-    # Split into london and rest of England
-    gdf_london = gdf[gdf["is_london"] == True]
-    gdf_roe = gdf[gdf["is_london"] == False]
-
-    # set common value limits for colour scale
-    default_plot_options = {
-        'vmin': df[column].min(),
-        'vmax': df[column].max(),
-        'edgecolor': "black",
-        'linewidth': 0.1,
-        'cmap': "OrRd",
-    }
-
-    if plot_options is None:
-        plot_options = {}
-
-    for k, v in default_plot_options.items():
-        if k not in plot_options:
-            plot_options[k] = v
-
-    def plot(gdf, ax, title="", legend=True):
-        gdf.plot(
-            ax=ax,
-            column=column,
-            legend=legend,
-            **plot_options
-        )
-        ax.set_aspect(1.63)
-        if title:
-            ax.set_title(title, size=14)
-        ax.axis("off")
-
-    fig = plt.gcf()
-    if not subplot_spec:
+
+    def __init__(self, description=None, author="EBM DataLab"):
+        """Map Loader is initiated with description and author for clarity
+
+            Args:
+                description (str): Description at a broad level of the dataset
+                author (str): Author - set as EBM Datalab if left blank
+
+        """
+
+        self.description = description
+        self.author = author
+
+        self.date = datetime.datetime.now()
+        #Datetime (obj): Automatically records datetime as time class is initiated
```

Comment:
Don't bother changing this now as it's super-trivial but just for future reference, the convention is that comments go above the line they refer to so as to match natural reading order. (Doc strings are a bit of a weird exception in this regard because they have to go inside the function they document.)